sdk/servicebus/service-bus/samples-dev/topicSubscriptionWithRuleOperationsSample.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * This sample demonstrates the topic filter functionality in Service Bus.
+ * It follows similar functionality to TopicSubscriptionWithRuleOperationsSample/program.cs
+ *
+ * https://github.com/Azure/azure-service-bus/blob/master/samples/DotNet/GettingStarted/Microsoft.Azure.ServiceBus/TopicSubscriptionWithRuleOperationsSample/Program.cs
+ *
+ * @summary Demonstrates how to filter messages in Service Bus
+ * @azsdk-weight 100
+ */
+
+import {
+  ServiceBusClient,
+  ServiceBusAdministrationClient,
+  ServiceBusMessage,
+} from "@azure/service-bus";
+import { DEFAULT_RULE_NAME } from "../src/util/constants";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// Define connection string and related Service Bus entity names here
+const connectionString = process.env.SERVICEBUS_CONNECTION_STRING || "<connection string>";
+// const queueName = process.env.QUEUE_NAME || "<queue name>";
+const topicName = "TopicSubscriptionWithRuleOperationsSample";
+
+const firstSetOfMessages: ServiceBusMessage[] = [
+  { subject: "Red", body: "test-red1" },
+  { subject: "Red", body: "test-red2", correlationId: "notimportant" },
+  { subject: "Red", body: "test-red3", correlationId: "important" },
+  { subject: "Blue", body: "test-blue1" },
+  { subject: "Blue", body: "test-blue2", correlationId: "notimportant" },
+  { subject: "Blue", body: "test-blue3", correlationId: "important" },
+  { subject: "Green", body: "test-green1" },
+  { subject: "Green", body: "test-green2", correlationId: "notimportant" },
+  { subject: "Green", body: "test-green3", correlationId: "important" },
+];
+
+const NoFilterSubscriptionName = "NoFilterSubscription";
+const SqlFilterOnlySubscriptionName = "RedSqlFilterSubscription";
+const SqlFilterWithActionSubscriptionName = "BlueSqlFilterWithActionSubscription";
+const CorrelationFilterSubscriptionName = "ImportantCorrelationFilterSubscription";
+
+export async function main() {
+  const sbClient = new ServiceBusClient(connectionString);
+  const sbAdminClient = new ServiceBusAdministrationClient(connectionString);
+
+  await sbAdminClient.createTopic(topicName);
+  await sbAdminClient.createSubscription(topicName, NoFilterSubscriptionName);
+  await sbAdminClient.createSubscription(topicName, SqlFilterOnlySubscriptionName);
+  await sbAdminClient.createSubscription(topicName, SqlFilterWithActionSubscriptionName);
```

Comment:
`SqlFilterWithActionSubscriptionName` is used in `createSubscription()` three times.